### PR TITLE
Add pod-selector kubectl drain

### PR DESCRIPTION
**Pull request by juanvallejo**
**Release note**:
```release-note
Added the ability to select pods in a chosen node to be drained, based on given pod label-selector
```

This patch adds the ability to select pods in a chosen node to be drained, based on given pod label-selector. Related downstream issue: https://github.com/openshift/origin/issues/17554

Further, it removes explicit, specific, pod-controller check. The `drain` command currently fails if a pod has a controller of a `kind` [not explicitly handled in the command itself](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/drain.go#L331). This causes `drain` to be unusable if a node contains pods managed by third-party, or "unknown" controllers.

Based on [this comment](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/drain.go#L353), the expectation was to fail if a pod's controller was not found for whatever reason. I believe that the `drain` command should not care about the existence of a pod controller. It should only care whether a pod has one, and act according to that controller kind. This solves a downstream bug: https://github.com/openshift/origin/issues/17563

cc @fabianofranz @deads2k @kubernetes/sig-cli-misc

### CODE REVIEW

**Comment by juanvallejo** [comment]
ping @fabianofranz or @soltysh

**Comment by juanvallejo** [comment]
@fabianofranz friendly ping

File: pkg/kubectl/cmd/drain.go [hunk 1]
```diff
@@ -328,37 +338,12 @@ func (o *DrainOptions) deleteOrEvictPodsSimple(nodeInfo *resource.Info) error {
...
-		return o.client.BatchV1().Jobs(namespace).Get(controllerRef.Name, metav1.GetOptions{})
-	case "ReplicaSet":
-		return o.client.ExtensionsV1beta1().ReplicaSets(namespace).Get(controllerRef.Name, metav1.GetOptions{})
-	case "StatefulSet":
-		return o.client.AppsV1beta1().StatefulSets(namespace).Get(controllerRef.Name, metav1.GetOptions{})
-	}
-	return nil, fmt.Errorf("Unknown controller kind %q", controllerRef.Kind)
-}
-
 func (o *DrainOptions) getPodController(pod corev1.Pod) (*metav1.OwnerReference, error) {
```
Comment (deads2k) [inline pkg/kubectl/cmd/drain.go]: This method no longer returns an error.

**Comment by deads2k** [comment]
Fix method signature, squash in your tests, then lgtm.

File: pkg/kubectl/cmd/drain.go [hunk 2]
```diff
@@ -368,18 +348,14 @@ func (o *DrainOptions) unreplicatedFilter(pod corev1.Pod) (bool, *warning, *fata
 		return true, nil, nil
 	}
 
-	controllerRef, err := o.getPodController(pod)
-	if err != nil {
+	controllerRef := o.getPodController(pod)
```
Comment (deads2k) [inline pkg/kubectl/cmd/drain.go]: Is this just?

```go
if controllerRef != nil{
    return true, nil, nil
}
if o.Force{
    return true, &warning{kUnmanagedWarning}, nil
}

return false, nil, &fatal{kUnmanagedFatal}
```

File: pkg/kubectl/cmd/drain.go [hunk 3]
```diff
@@ -393,23 +369,20 @@ func (o *DrainOptions) daemonsetFilter(pod corev1.Pod) (bool, *warning, *fatal)
 	// The exception is for pods that are orphaned (the referencing
 	// management resource - including DaemonSet - is not found).
 	// Such pods will be deleted if --force is used.
-	controllerRef, err := o.getPodController(pod)
-	if err != nil {
+	controllerRef := o.getPodController(pod)
```
Comment (deads2k) [inline pkg/kubectl/cmd/drain.go]: Use if's and early returns please

**Comment by deads2k** [comment]
comments and squash in that last commit.

**Comment by juanvallejo** [comment]
@deads2k thanks, done

**Pull request merged**